eclipse-collections/src/main/java/org/eclipse/collections/impl/collector/Collectors2.java

```diff
@@ -878,6 +880,29 @@ private Collectors2()
                 EMPTY_CHARACTERISTICS);
     }
 
+    /**
+     * Groups the elements using the {@code groupBy} function and all the elements that map to the same key are
+     * aggregated together using the {@code aggregator} function. The third parameter, the {@code zeroValueFactory}
+     * function, creates the initial value in each aggregation. Aggregate results are allowed to be immutable as they
+     * will be replaced in the map.
+     */
+    public static <T, K, R extends MutableMap<K, T>> Collector<T, ?, R> aggregateBy(
+            Function<? super T, ? extends K> groupBy,
+            Function0<? extends T> zeroValueFactory,
+            Function2<? super T, ? super T, ? extends T> aggregator,
+            Supplier<R> supplier)
```

Comment:
Why do we need to pass in the supplier if the result is always MutableMap?  If we keep the supplier, it would be good to have a test showing how a different MutableMap is used.  It would be even better if R extends MapIterable was the return type as it is on RichIterable.  This would allow then for a Supplier that can return a MutableSortedMap, which is not in the MutableMap hierarchy.

Reply:
If we use MapIterable, I am not sure how to combine the results in case of parallel stream. Because both r1 and r2 would be of type R extends MapIterable and how do we add elements from r2 to r1?

Reply:
Fair point, I understand now.  Ok, the signature makes sense.

Reply:
Shouldn't it be MutableMapIterable?

Reply:
MutableMapIterable would work, I believe. We can use suppliers returning different MutableMap implementations.